client/components/dashboard.js

```diff
@@ -1,45 +1,97 @@
 import React, {Component} from "react";
+import Home from './home';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {Link} from 'react-router';
+import GroupActions from '../actions/group';
+import SideNav from './sidenav';
+import ReactPaginate from 'react-paginate';
 
+const LIMIT = 6;
+class Dashboard extends Component {
+  setGroupIDHandler(groupID, groupName) {
+    localStorage.setItem('groupID', groupID);
+    localStorage.setItem('groupName', groupName);
+  }
+  componentDidMount() {
+    this.props.getTotalGroups(JSON.parse(localStorage.user).user)
+    if (localStorage.groupOffset) {
+      this.props.getGroups(JSON.parse(localStorage.user).user, localStorage.groupOffset, LIMIT);
+    } else {
+      this.props.getGroups(JSON.parse(localStorage.user).user, 0, LIMIT);
+    }
+  }
+
+  handlePageClick(data) {
+    let selected = data.selected;
+    let offset = Math.ceil(selected * LIMIT);
+    localStorage.setItem('groupOffset', offset);
+    this.props.getGroups(JSON.parse(localStorage.user).user, offset, LIMIT);
+  };
 
-export default class Dashboard extends Component{
-  render(){
+  render() {
+    let dashboard;
+    let pagination;
+    if (!localStorage.user) {
+      dashboard = <Home/>;
+    } else {
+      if (this.props.usersGroups.reqProcessing) {
```

Comment:
I see lot of `this.props.something.somethingElse` in your code. Try using es6 destructuring e.g` const { usersGroups, getGroups } = this.props`, you can then use `userGroups.reqProcessing`. Read more about it.